usageReports/rds/get_rds_data.go

```diff
@@ -0,0 +1,162 @@
+	l := jsonlog.LoggerFromContextOrDefault(ctx)
+	index := strings.Join(parsedParams.indexList, ",")
+	searchService := GetElasticSearchRdsDailyParams(
+		parsedParams,
+		es.Client,
+		index,
+	)
+	res, err := searchService.Do(ctx)
+	if err != nil {
+		if elastic.IsNotFound(err) {
+			l.Warning("Query execution failed, ES index does not exists : "+index, err)
+			return nil, http.StatusOK, err
+		}
+		l.Error("Query execution failed : "+err.Error(), nil)
+		return nil, http.StatusInternalServerError, fmt.Errorf("could not execute the ElasticSearch query")
+	}
+	return res, http.StatusOK, nil
+}
+
+// makeElasticSearchRdsMonthlyRequest prepares and run the request to retrieve a month report
+// based on the esQueryParams
+// It will return the data, an http status code (as int) and an error.
+// Because an error can be generated, but is not critical and is not needed to be known by
+// the user (e.g if the index does not exists because it was not yet indexed ) the error will
+// be returned, but instead of having a 500 status code, it will return the provided status code
+// with empty data
+func makeElasticSearchRdsMonthlyRequest(ctx context.Context, parsedParams rdsQueryParams) (*elastic.SearchResult, int, error) {
+	l := jsonlog.LoggerFromContextOrDefault(ctx)
+	index := strings.Join(parsedParams.indexList, ",")
+	searchService := GetElasticSearchRdsMonthlyParams(
+		parsedParams,
+		es.Client,
+		index,
+	)
+	res, err := searchService.Do(ctx)
+	if err != nil {
+		if elastic.IsNotFound(err) {
+			l.Warning("Query execution failed, ES index does not exists : "+index, err)
+			return nil, http.StatusOK, err
+		}
+		l.Error("Query execution failed : "+err.Error(), nil)
+		return nil, http.StatusInternalServerError, fmt.Errorf("could not execute the ElasticSearch query")
+	}
+	return res, http.StatusOK, nil
+}
+
+// getRdsDailyData gets RDS daily reports and parse them based on query params
+func getRdsDailyData(ctx context.Context, params rdsQueryParams, user users.User, tx *sql.Tx) (int, []rds.Report, error) {
+	searchResult, returnCode, err := makeElasticSearchRdsDailyRequest(ctx, params)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	accountsAndIndexes, returnCode, err := es.GetAccountsAndIndexes(params.accountList, user, tx, es.IndexPrefixLineItems)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	params.accountList = accountsAndIndexes.Accounts
+	params.indexList = accountsAndIndexes.Indexes
+	costResult, _, _ := makeElasticSearchCostRequest(ctx, params)
+	res, err := prepareResponseRdsDaily(ctx, searchResult, costResult)
+	if err != nil {
+		return http.StatusInternalServerError, nil, err
+	}
+	return http.StatusOK, res, nil
+}
+
+// getRdsData gets RDS monthly reports based on query params, if there isn't a monthly report, it calls getRdsDailyData
+func getRdsData(request *http.Request, parsedParams rdsQueryParams, user users.User, tx *sql.Tx) (int, []rds.Report, error) {
```

Comment:
This function should be named `getRdsMonthlyData` in accordance with the function `getRdsDailyData`